qf_lib/data_providers/prefetching_data_provider.py

```diff
@@ -78,6 +79,10 @@ def __init__(self, data_provider: DataProvider,
 
         data_array = data_provider.get_price(all_tickers, fields, start_date, end_date, frequency)
 
+        if isinstance(data_provider, ExchangeRateProvider):
+            self.get_last_available_exchange_rate = lambda base_currency, quote_currency, frequency: \
+                data_provider.get_last_available_exchange_rate(base_currency, quote_currency, frequency)
```

Comment:
assigning lambda like this feels a bit off, I think I'd prefer to set self._data_provider = data_provider and have it as a normal function with isinstance check inside